src/middleware/PostQuery.ts

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+/**
+ *  This class will be for handling the transfer of the res.locals.graphqlGate object
+ *  to the webapp's backend
+ */
+export default class PostQuery {
+    // address to our webapp's backend, currently sits at localhost:3000
+    private gateURI: string;
+
+    // points to user's project in our webapp DB
+    private projectID: string;
+
+    // passed in by our logger middleware from res.locals.graphqlGate
+    private queryData: QueryData;
+
+    constructor(gateURI: string, projectID: string, queryData: QueryData) {
+        this.gateURI = gateURI;
+        this.projectID = projectID;
+        this.queryData = queryData;
+    }
+
+    // takes data from res.locals.graphqlGate and posts to webapp backend
+    public async post() {
+        const { complexity, timestamp, tokens } = this.queryData;
+
+        if (complexity < 0 || timestamp < 0 || tokens < 0)
```

Comment:
For later discussion. How intrusive do we want the logger to be? In a world where our rate limiter works perfectly, these would never be negative. In the case that the rate limiter has a bug, does it make sense for the logger to disrupt the application / middleware chain? Whether or not we throw an error or even log the request is up for discussion. Let's create an issue to discuss this afterwards.